src/preamble.js

```diff
@@ -2401,7 +2403,13 @@ function integrateWasmJS(Module) {
   Module['asmPreload'] = Module['asm'];
 
   // Memory growth integration code
+
+  var defaultReallocBuffer = Module['reallocBuffer'];
+
   Module['reallocBuffer'] = function(size) {
+    if (finalMethod === 'asmjs') {
+      return defaultReallocBuffer(size);
```

Comment:
Good point, this could be clearer. I refactored it to have asmjsReallocBuffer and wasmReallocBuffer, as you suggested. We can't assign the right one immediately, we only know which method succeeded later, but the refactor lets `Module['reallocBuffer']` be basically just calling the right one of the two, which seems a lot cleaner.